src/utils/knowledgeartifacts.js

```diff
@@ -55,6 +57,9 @@ function fetchEndpoint(url, planDefinition) {
     if (endpoint) {
       db.upsert(ENDPOINTS, endpoint, r => compareUrl(r.fullUrl, endpoint.fullUrl));
       debug(`Endpoint/${endpoint.id} saved to db`);
+
+      const baseUrl = endpoint.address.split('/$process-message')[0];
```

Comment:
Specifically looking for feedback to see if this is suffice

Reply:
Only thing I would add is a check to only try to register if the server isn't already in the servers collection. (or if it doesn't already have a client_id or something). I don't see that in the registerServer function but I think it would be fine either right here in fetchEndpoint or there in registerServer